test/internet/test-http2-update-settings.js

```diff
@@ -0,0 +1,92 @@
+
+const assert = require('assert');
+const http2 = require('http2');
+
+testUpdateSecureServerSettings();
+testUpdateServerSettings();
+
+function testUpdateSecureServerSettings() {
+  const server = http2.createSecureServer();
+  const newServerSettings = {
+    'headerTableSize': 1,
+    'initialWindowSize': 1,
+    'maxConcurrentStreams': 1,
+    'maxHeaderListSize': 1,
+    'maxFrameSize': 16385,
+    'enablePush': false,
+    'enableConnectProtocol': true
+  };
+  const faultyServerSettings = {
+    'headerTableSize': 'foo'
+  };
+  const currentServerSettings = getServerSettings(server);
+  assert.notDeepStrictEqual(currentServerSettings, newServerSettings);
+  server.updateSettings(newServerSettings);
+  const updatedServerSettings = getServerSettings(server);
+  assert.deepStrictEqual(updatedServerSettings, newServerSettings);
+  assert.throws(server.updateSettings(faultyServerSettings),
+                invalidValueHelper);
+  assert.throws(server.updateSettings(),
+                invalidObjectHelper);
+  assert.throws(server.updateSettings(undefined),
+                (exception) => invalidObjectHelper(exception, undefined));
+}
+
+function testUpdateServerSettings() {
+  const server = http2.createServer();
+  const newServerSettings = {
+    'enablePush': false
+  };
+  const faultyServerSettings = {
+    'headerTableSize': 'foo'
+  };
+  const currentServerSettings = getServerSettings(server);
+  assert.notDeepStrictEqual(currentServerSettings, newServerSettings);
+  server.updateSettings(newServerSettings);
+  const updatedServerSettings = getServerSettings(server);
+  assert.deepStrictEqual(updatedServerSettings, newServerSettings);
+  assert.throws(server.updateSettings(faultyServerSettings),
+                invalidValueHelper);
+  assert.throws(server.updateSettings(),
+                invalidObjectHelper);
+  assert.throws(server.updateSettings(undefined),
+                (exception) => invalidObjectHelper(exception, undefined));
+}
+
+function getServerSettings(server) {
+  return Object
+  .getOwnPropertySymbols(server)
+  .find((s) => s.toString() === 'Symbol(kOptions)')
+  .settings;
+}
+
+function invalidObjectHelper(exception, obj) {
+  assert.strictEqual(exception.constructor.name,
+                     'Error');
+  assert.strictEqual(exception.code,
+                     'ERR_HTTP2_INVALID_SETTING_OBJECT');
+  if (obj) {
+    assert.strictEqual(exception.nessage,
+                       `Invalid setting object "${obj}"`);
+  } else {
+    assert.strictEqual(exception.message,
+                       'The setting object cannot be empty');
+  }
+}
+
+function invalidValueHelper(exception) {
+  assert.strictEqual(exception.constructor.name,
+                     'TypeError');
+  assert.strictEqual(exception.code,
+                     'ERR_HTTP2_INVALID_SETTING_VALUE');
+  assert.strictEqual(exception.message,
+                     'Invalid value for setting "headerTableSize": foo');
+}
```

Comment:
This test doesn't connect to anything on the internet, does it? If so, can you move it to `test/parallel` instead of `test/internet`? `test/internet` is for our tests that require a network connection to the internet. We try to put only tests that need to go in `test/internet` there because those tests do not get run during a typical CI run.